ext/standard/tests/math/round_gh12143_2.phpt

```diff
@@ -0,0 +1,27 @@
+--TEST--
+GH-12143: Test rounding of 0.50000000000000011.
+--FILE--
+<?php
+foreach ([
+    0.50000000000000011,
+    -0.50000000000000011,
+] as $number) {
+    foreach ([
+        'PHP_ROUND_HALF_UP',
+        'PHP_ROUND_HALF_DOWN',
+        'PHP_ROUND_HALF_EVEN',
+        'PHP_ROUND_HALF_ODD',
+    ] as $mode) {
+        printf("%-20s: %+.17g -> %+.17g\n", $mode, $number, round($number, 0, constant($mode)));
+    }
+}
+?>
+--EXPECT--
+PHP_ROUND_HALF_UP   : +0.50000000000000011 -> +1
+PHP_ROUND_HALF_DOWN : +0.50000000000000011 -> +1
+PHP_ROUND_HALF_EVEN : +0.50000000000000011 -> +1
+PHP_ROUND_HALF_ODD  : +0.50000000000000011 -> +1
+PHP_ROUND_HALF_UP   : -0.50000000000000011 -> -1
+PHP_ROUND_HALF_DOWN : -0.50000000000000011 -> -1
+PHP_ROUND_HALF_EVEN : -0.50000000000000011 -> -1
+PHP_ROUND_HALF_ODD  : -0.50000000000000011 -> -1
```

Comment:
[nits]

Is'nt it `0.5000000000000001`?

`0.50000000000000011` cannot be represented in double precision.

Reply:
```
#include <math.h>
#include <stdio.h>

int
main(void) {
	printf("%.17g\n", nextafter(0.5, 0));
	printf("%.17g\n", nextafter(0.5, 1));
}
```

Results in:

```
0.49999999999999994
0.50000000000000011
```

And as you can see in the output, `0.50000000000000011` can be represented exactly. That's why the test also dumps the original number.

```
#include <math.h>
#include <stdio.h>

int
main(void) {
	printf("%.17g\n", 0.5000000000000001);
}
```

results in `0.50000000000000011`.

Reply:
Oops, excuse me, it seems the following result was confusing me.

```
var_dump(0.500000000000000011);
// float(0.5000000000000001)
```

Reply:
The output of `var_dump()` depends on the `precision` ini setting. To losslessly print a double you need 17 digits as per https://stackoverflow.com/a/21162120/782822. That's why the test uses `%.17g`.

Reply:
Yes, I mistakenly thought that `serialize_precision = -1` would output 17 digits. Sorry for bothering you.